### ISSUE TITLE
Feature: specify delay before taking the snapshot

### ISSUE THREAD

**Issue by joscha** [percy/react-percy#56]
Is there a way to specifiy a delay before a storybook run is rendered? We have some async things going on and would like to add a tiny delay - is this option potentially available on the Percy API and I could incorporate a flag into #50?

**Comment by timhaines** [comment]
Hi @joscha - we do have a delay of 2 seconds in place before the screenshot is taken.  It sounds like that's not enough - but I suspect you would want your components to be fast enough that it was enough?

Possibly we could look at exposing a param that allows an extra 1 - 3 seconds.  Let me discuss this internally and get back to you.

**Comment by fotinakis** [comment]
Joshcha — it sounds like the delay we have (which is mostly an internal-only thing for a special case) is not currently handling your situation. I doubt additional delay would actually help here — what are you seeing in your snapshots not render in time?

**Comment by joscha** [comment]
We are seeing some images not rendering in time, the sequence is:

i.e. The sequence is

1. A new image bundle is added to a resolver which kicks off an HTTP request for the image
2. The story renders (without an image in those elements referencing the image above)
3. The image response returns
4. The resolver updates it internal store of images, which causes MobX reactions to execute
5. The story re-renders to include the image

It should be faster than 2 seconds, I think.

If you look at build 317381 you can see an example of the flaky image.

**Comment by joscha** [comment]
When are the two seconds measured? From window onload or total load time?

**Comment by fotinakis** [comment]
Hey joscha, we'd rather people not rely on the 2 second buffer, it's really an internal implementation detail that we might change. For example, we likely won't keep it as we implement support for new browsers like Chrome. But to answer your question, I believe it's after the DOMContentLoaded event fires.

**Comment by joscha** [comment]
I understand. I am wondering about the best way to make this work. If its `DOMContentLoaded` we could link the images in the main document, so they would be loaded and cached by the time that event fires.

**Comment by timhaines** [comment]
@joscha Checking in on this - is it still an open issue / ongoing problem for you?

**Comment by joscha** [comment]
Yes, we are still unsure why it happens. There is an `onload` event on the
image in question - if I change that to come back immediately it fixes the
problem. A bit odd is that 2 seconds after `DOMContentLoaded` should be
more than enough.

**Comment by zleight1** [comment]
I see I'm a bit late to the party here but it looks like the storybook is still using the this deprecated react client and while 2 seconds should be plenty for mounting our components I'm still getting false matches.

See: https://github.com/percy/percy-storybook/issues/220

Was this exposed in the new client API (percy-js)?

**Comment by felixfbecker** [comment]
Is the default delay also applied when the width is switched? I am observing that my largest size is screenshotting fine, but my smaller one seems to not have been able to adjust to the smaller screensize yet (there is some JavaScript layout going on). It's impossible to observe when resizing the browser manually though, leading me to believe that Percy changes the width of the browser, then instantly takes a snapshot.
